### ISSUE TITLE
[BUG] Class method return types needed for >8.0 support

### ISSUE THREAD

**Issue by sosherof** [Bandwidth/php-sdk#62]
In 8.1 and above, class methods based on an implemented class must have a return type.   Without these, PHP will throw a return type mismatch warning or error.

The following code is from Models\BandwidthMessage class, how it is today:

```php
class BandwidthMessage implements \JsonSerializable
{
    public function jsonSerialize()
    {
    $json = array();
    //some other code here
    return array_filter($json);
    }
}
``` 

vs. how it must be for PHP 8.1 and above:

```php
class BandwidthMessage implements \JsonSerializable
{
    public function jsonSerialize(): array
    {
    $json = array();
    //some other code here
    return array_filter($json);
    }
}
```

There are multiple classes in the Bandwidth SDK that "implements" JsonSerializable.  There may be other things that make the SDK not fully 8.1 and above compatible but this one stood out as I tested the Messaging SDK under PHP 8.2.5.

See https://www.php.net/manual/en/migration81.incompatible.php

**Comment by ajrice6713** [comment]
Hey @sosherof - thanks for opening this issue. 

I have some work on our backlog to look at updating this SDK to be compatible with PHP >= 8.1 - cant promise a specific timeline right now - but we will look into this in the next couple of days and get back to you! 

For some context - this SDK was previously autogenerated + is now maintained manually & we are currently working to migrate to the [OpenAPI Generator](https://github.com/OpenAPITools/openapi-generator) project to improve SDK generation.

**Comment by ajrice6713** [comment]
Hey @sosherof - I have PR open to get this resolved, #64, if you wanted to take a look before we push this out! I think these changes will resolve your issue.

**Comment by sosherof** [comment]
> Hey @sosherof - I have PR open to get this resolved, #64, if you wanted to take a look before we push this out! I think these changes will resolve your issue.

Looking good.  I made similar changes, though not as thorough, to get me by for now.  I only use the Messaging components out of this SDK so I haven't looked at the other files but one thing to look out for is methods that implement functions that return "mixed".  This is a valid type in 8.0 but not in 7.4.  In another library I tweaked (httpful) to work on both 7.4 and 8.2, I gave up and used the "#[\ReturnTypeWillChange]" attribute right above the function definition.  In this case, the class had implemented \ArrayAccess (offsetGet, offsetSet, offsetUnset).  In PHP 8.0, ArrayAccess methods use the return type "mixed".

When do you think you'll merge this into the next rev?

**Comment by ajrice6713** [comment]
Thanks for pointing that out - Our unit tests on 7.4 are passing at the moment - but i dont trust the coverage enough, so let me make sure that we have no mixed types (i dont think we do) and get a review from my team, and we should be releasing this next Tuesday (5/23)!

**Comment by ajrice6713** [comment]
Hey @sosherof - this fix is included in v10.4.0

Let me know if you have any questions!
